src/textual/widgets/_list_view.py

```diff
@@ -0,0 +1,145 @@
+from __future__ import annotations
+
+import asyncio
+from asyncio import Future
+
+from textual import events
+from textual.binding import Binding
+from textual.containers import Vertical
+from textual.geometry import clamp
+from textual.message import Message
+from textual.reactive import reactive
+from textual.widgets._list_item import ListItem
+
+
+class ListView(Vertical, can_focus=True, can_focus_children=False):
+    DEFAULT_CSS = """
+    ListView {
+        scrollbar-size-vertical: 1;
+    }
+    """
+
+    BINDINGS = [
+        Binding("down", "down", "Down"),
+        Binding("up", "up", "Up"),
+        Binding("enter", "select", "Select"),
+    ]
+
+    index = reactive(0, always_update=True)
+
+    def __init__(
+        self,
+        *children: ListItem,
+        initial_index: int | None = 0,
+        name: str | None = None,
+        id: str | None = None,
+        classes: str | None = None,
+    ) -> None:
+        super().__init__(*children, name=name, id=id, classes=classes)
+        self.index = initial_index
+
+    @property
+    def highlighted_child(self) -> ListItem | None:
+        """Get the currently highlighted ListItem
+
+        Returns:
+            ListItem | None: The currently highlighted ListItem, or None if nothing is highlighted.
+        """
+        if self.index is None:
+            return None
+        elif 0 <= self.index < len(self.children):
+            return self.children[self.index]
+
+    def validate_index(self, index: int | None) -> int | None:
+        if not self.children or index is None:
+            return None
+        return self._clamp_index(index)
+
+    def _clamp_index(self, index: int) -> int:
+        """Clamp the index to a valid value given the current list of children"""
+        last_index = max(len(self.children) - 1, 0)
+        return clamp(index, 0, last_index)
+
+    def _is_valid_index(self, index: int | None) -> bool:
+        """Return True if the current index is valid given the current list of children"""
+        if index is None:
+            return False
+        return 0 <= index < len(self.children)
+
+    def watch_index(self, old_index: int, new_index: int) -> None:
+        if self._is_valid_index(old_index):
+            old_child = self.children[old_index]
+            old_child.highlighted = False
+        if self._is_valid_index(new_index):
+            new_child = self.children[new_index]
+            new_child.highlighted = True
+        else:
+            new_child = None
+
+        self._scroll_highlighted_region()
+        self.emit_no_wait(self.Highlighted(self, new_child))
+
+    async def append(self, item: ListItem) -> None:
```

Comment:
Could we make this non-async?

Use `emit_no_wait` and return the awaitable from the `mount` method.

Hopefully that won't break anything.

Ditto for `clear`

Reply:
Technically yes, but I think this will cause a lot confusion/subtle bugs. It needs to be async because we need to `await` the call to `remove` inside it. Otherwise we'll send a `ChildrenUpdated` with undefined `children` (could contain children before or after update, who knows).

If I receive a `ChildrenUpdated` message, I won't know inside my `on_children_updated` handler if the children have actually been updated yet or not, and I'll have no easy way of `await`ing it (I won't have access to the `AwaitRemove` in there). It would mean we can't actually trust that message any more since the state of the children is undefined. Doing anything to my ListView on receipt of this message would be dangerous.

Reply:
Do we need a children update message? Generally it is the parent that is doing the updating, so it will know there has been an updated.

Do you have a use-case in mind?

Reply:
I've removed it, don't think it's necessary.

It was to give the ancestors a single hook (the children update handler) that they could use when the children changed. e.g. a single place they can update a label to reflect the number of items in the list rather.